File: 2026-2/sesion-de-clase-08/softprog/softprog-modelo/src/main/java/pe/edu/pucp/softprog/modelo/Registro.java
package pe.edu.pucp.softprog.modelo;

public abstract class Registro {
    private int id;
    private boolean activo;

    public Registro() {
    }

    public Registro(final Registro registro) {
        if  (registro == null) {
            throw new IllegalArgumentException("registro no puede ser nulo");
        }
        setId(registro.getId());
        setActivo(registro.isActivo());
    }

    public int getId() {
        return id;
    }

    public void setId(int id) {
        if (id < 0) {
            throw new IllegalArgumentException("id no puede ser negativo");
        }
        this.id = id;
    }

    public boolean isActivo() {
        return activo;
    }

    public void setActivo(boolean activo) {
        this.activo = activo;
    }

    @Override
    public String toString() {
        return String.format("%-5d\t%-5b", id, activo);
    }
}

File: 2026-2/sesion-de-clase-11/SoftProg/SoftProg.Modelo/Registro.cs
namespace SoftProg.Modelo {
    public abstract class Registro {
        public Registro() { }

        public Registro(Registro registro) { 
            Id = registro.Id;
            IsActivo = registro.IsActivo;
        }

        public int Id { get; set; }
        public bool IsActivo { get; set; }

        public override string ToString() {
            return $"Id: {Id}, Activo: {IsActivo}, ";
            ;
        }

    }
}
